packages/core/ui-settings/core-ui-settings-server-internal/src/clients/base_ui_settings_client.ts

```diff
@@ -99,6 +100,20 @@ export abstract class BaseUiSettingsClient implements IUiSettingsClient {
     }
   }
 
+  private async getDefaultValues(context?: GetUiSettingsContext) {
+    const values: { [key: string]: unknown } = {};
+
+    for (const [key, definition] of Object.entries(this.defaults)) {
```

Comment:
it might be worth doing this async? Imagine we have more dynamic settings with getValue() . Then now they all will be resolving sequentially blocking each other

Reply:
You mean doing it in parallel? Yes that would be better 👍

Reply:
Done in https://github.com/elastic/kibana/pull/187763/commits/56a895180a632d19a710ed2829057ad04ff67a5b